### PR TITLE
Unit 2 michael egner

### PR DESCRIPTION
**Pull request by MEgner410**
# Submitting Unit 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### ✋ Reminder

Do NOT merge your assessment!

### 📝 Feedback

If you would like specific feedback please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -27,26 +29,92 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+    if (typeof number !== 'number') {
+        return "not a valid number";
+    }
+    if (number % 3 === 0) {
+        return `${number} is divisible by three`
+    } else {
+        return `${number} is not divisible by three`;
+    }
+    }
+ // const object1 = { number: 15 }
```
Comment (sjproctor) [inline code-challenges.test.js:45]: Interesting approach with the destructuring. It also looks like you did two separate conditional statements to handle an edge case.

File: code-challenges.test.js [hunk 2]
```diff
@@ -27,26 +29,92 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+    it('should return "not a valid number" for invalid input', () => {
+        const result = isDivisibleByThree({ number: 'invalid' });
+        expect(result).toEqual("not a valid number");
+    })
+}); 
 
 // Pseudo code:
+// input single array of numbers
+// output array with numbers multipled by 3, - 0, 15, and -7
+// does something to everything in the array value * 3
```
Comment (sjproctor) [inline code-challenges.test.js:77]: I don't think this is the correct pseudo code for the problem you are solving.

File: code-challenges.test.js [hunk 3]
```diff
@@ -27,26 +29,92 @@ const numbersArray2 = [24, 27, 30, 33, 36]
...
+const RandomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+
+const uppercaseWords1 = randomNouns1.map(word => word.toUpperCase())
+const uppercaseWords2 = randomNouns2.map(word => word.toUpperCase())
+
+
+// Pseudo code:
+//input: make a array that is empty store capitalized words
+// output: return the random words with the first letter in the array capitalized.
+
```
Comment (sjproctor) [inline code-challenges.test.js:119]: Remember to write your test!

File: coding-concepts.js [hunk 4]
```diff
@@ -21,29 +22,32 @@ const greeter = (name) => {
...
+//   frameworks: ["React", "Ruby on Rails"],
+//   databases: "PostgreSQL",
+//   versionControl: "GitHub"
+// }
 // console.log(myCodingSkills.languages[0])
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: JavaScript
+// b) Verify and explain: the statement prints the first element of index
```
Comment (sjproctor) [inline coding-concepts.js:50]: There are two pieces here. The first is getting the value of the language key. Then getting the element at the zeroth index.

**Review by sjproctor — APPROVED** [review]
Nice job, Michael. I really encourage you to try and follow the test-driven development practice of creating your test, watching the test fail, then creating the code. Make sure to use pseudo code to explain your thought process and dive deeper into your code. The learning is in the discovery of what your code is doing. This assessment is at a novice level of skill development.

**Comment by MEgner410** [comment]
Thank you this was a tough one for me to be honest
